utils/datasets.py

```diff
@@ -646,6 +655,52 @@ def collate_fn4(batch):
 
 
 # Ancillary functions --------------------------------------------------------------------------------------------------
+def json_paths2mask(self, json_dir: str):
```

Comment:
It looks like a function, not a method of one class. I don't know why there is an argument `self`, and it is not used in this function